pkg/reconciler/v1alpha1/pipelinerun/pipelinerun_test.go

```diff
@@ -1122,3 +1301,79 @@ func TestReconcilePropagateAnnotations(t *testing.T) {
 		t.Errorf("expected to see TaskRun %v created. Diff %s", expectedTaskRun, d)
 	}
 }
+
+func TestReconcileWithConditionChecks(t *testing.T) {
```

Comment:
Nice, thanks! This test verifies the happy path in case of one conditions; it might be worth adding coverage for more kind of situations, like errors happening along the way and/or having multiple conditions.

Reply:
Added tests that tests reconcile with multiple conditions as well as when a condition fails (but there are other tasks to be run)